sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageSessionProcessor.java

```diff
@@ -33,19 +31,24 @@ public class MessageSessionProcessor extends ServiceBusScenario {
     private int prefetchCount;
 
     @Override
-    public void run() {
-        ServiceBusProcessorClient processor = getSessionProcessorBuilder(options)
+    public RunResult run() {
+        AtomicReference<RunResult> runResult = new AtomicReference<>(RunResult.INCONCLUSIVE);
+        ServiceBusProcessorClient processor = toClose(getSessionProcessorBuilder(options)
                 .maxConcurrentSessions(maxConcurrentSessions)
                 .maxConcurrentCalls(maxConcurrentCalls)
                 .prefetchCount(prefetchCount)
                 .processMessage(messageContext -> messageContext.complete())
                 .processError(err -> {
-                    throw LOGGER.logExceptionAsError(new RuntimeException(err.getException()));
+                    LOGGER.atError()
+                        .addKeyValue("source", err.getErrorSource())
+                        .log("processor error", err.getException());
+                    runResult.set(RunResult.ERROR);
                 })
-                .buildProcessorClient();
-
+                .buildProcessorClient());
+        toClose((AutoCloseable) () -> processor.stop());
```

Comment:
good catch! we already call `toClose` on the processor when it's created. But it seems stopping it is not necessary since we're going to close it anyway, so I'm going to remove this line.